Feature clear results

breadadams opened the pull request:
Close #3

breadadams committed:
Create initial components for “clear results”

+ Create “ClapWrap”, base wrap for the whole component
+ Create “ClearClaps”, side button to clear claps
+ Move & rename clap icon component

breadadams committed:
Update “color” stories primaryColor

(If primaryColor === secondaryColor, the ClearClaps “X” isn’t visible)

breadadams committed:
Update ClearClaps

+ Only display “clear claps” button when button has been clicked

breadadams committed:
Add `onClickClear` method

+ Add method to reset the users clicks

Kikobeats commented:
OMG! 😍 

some points before merge

At Medium they are adding a bit of delay after hide the Close button when the mouse is out of the component.

![screen recording 2018-03-05 at 09 50 am](https://user-images.githubusercontent.com/2096101/36965680-ade4a5fe-205a-11e8-841d-3be7b5f556d1.gif)

The Background color of the Close Icon should be use `primaryColor` and `secondaryColor` to be possible use custom color properly.

breadadams committed:
Update ClearClaps transition

+ Add react-transition-group dependency
+ Give state to `ClapWrap`
+ Use rtg’s `Transition` component in `ClearClaps`

breadadams committed:
Fix `onClickClear` method

+ Reset `count` to 0

breadadams committed:
Hide `ClapCountTotal` when total === 0

breadadams committed:
Merge branch 'master' into feature-clear-results

# Conflicts:
#	package.json
#	src/index.js
#	stories/index.js

Kikobeats commented on package.json:
```diff
@@ -26,7 +26,7 @@
     "polished": "~1.9.2",
     "react-transition-group": "^2.2.1",
-    "styled-components": "~3.1.4"
+    "styled-components": "~3.2.0"
```
careful because the current `3.2.0` is actually broken!

Kikobeats commented on src/components/ClapButton.js:
```diff
@@ -19,7 +19,7 @@ const ClapButton = styled.button`
   outline: 1px solid transparent;
   border-radius: 50%;
-  background: none;
+  background: #fff;
```
maybe are you thinking in `transparent`?

breadadams replied:
I had to set a background because you could see the ✖️ behind the circle
